### PR TITLE
Table

### PR DESCRIPTION
**Pull request by swfiua**
Allow edge colours for the cells making up the table to be supplied.

Using a lighter edge colour makes the numbers easier to read.

Fix auto_set_font_size to check height of text as well as width.

Use "center_baseline" as the vertical text alignment as it seems to centre better within cells.

### CODE REVIEW

File: lib/matplotlib/text.py [hunk 1]
```diff
@@ -1192,7 +1192,7 @@ def set_verticalalignment(self, align):
 
         ACCEPTS: [ 'center' | 'top' | 'bottom' | 'baseline' ]
```
Comment (phobson) [inline lib/matplotlib/text.py]: should we add `'center_baseline'` here too?

Comment (swfiua) [inline lib/matplotlib/text.py]: yes -- will do.

**Comment by swfiua** [comment]
I just took a look at the table tests.   The image based tests will need updating anyway.

The images will have changed subtly in some cases.

**Comment by story645** [comment]
can you please post some images with the new functionality?

**Comment by swfiua** [comment]
There will be some new images with the new tests -- once I have written them ;)

**Comment by swfiua** [comment]
I've added a test for the cellEdgeColour(s) options.

This commit https://github.com/matplotlib/matplotlib/pull/8303/commits/57c1aefba84392721a0e34c6a2e835357fd52e5d

Has an example (in the baseline_images for tests) that shows the use of edge colours.

The other part of this fix is to make the "auto_set_font_size" code a bit smarter and the use of 'center_baseline'  to align the text vertically.

Overall, the tables are more pleasing to the eye.

I fixed the examples/pylab_examples/table_demo.py  to use a light grey for the cell edge colours -- this makes it easier to pick out the cell text.

**Comment by NelleV** [comment]
I didn't have time to do a full review, but can you also make sure to not expose new functions to Matplotlib's API unless absolutely necessary? There seems to be a couple of new public functions that could be private from a first glance.

**Comment by swfiua** [comment]
@NelleV 

Re the API:  I'm not sure the table.Cell class makes it into the API, I've only ever used axes.table().

Cell.auto_set_font_size() has grown and extra (optional) hint argument, for the font size -- this is partly an attempt to make the font setting do less work by learning from the other cells that have been set already,

I added get_required_dimension() to return required width, height of the cell.   Previously the code only cared about the width, so could get by with just get_required_width()

**Comment by NelleV** [comment]
Closing this in favor of #8325

**Pull request closed without merging**